File: app/src/main/java/com/example/projectapp/utils/SharedViewModel.java
package com.example.projectapp.utils;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class SharedViewModel extends ViewModel {

    private final MutableLiveData<String> onNavigationFromBottomSheetToRegister = new MutableLiveData<>();
    private final MutableLiveData<Integer> onNavigationFromEmployeeDialog = new MutableLiveData<>();

    public void select(String item) {
        onNavigationFromBottomSheetToRegister.setValue(item);
    }

    public LiveData<String> getSelected() {
        return onNavigationFromBottomSheetToRegister;
    }

    public void navigationComplete() {
        onNavigationFromBottomSheetToRegister.setValue(null);
    }

    public void onDialogPositiveClick(int item) {
        onNavigationFromEmployeeDialog.setValue(item);
    }

    public LiveData<Integer> getDialogPositiveClicked() {
        return onNavigationFromEmployeeDialog;
    }

    public void dialogActionComplete() {
        onNavigationFromEmployeeDialog.setValue(null);
    }


}

File: app/src/main/java/com/example/projectapp/utils/SharedViewModel.kt
package com.example.projectapp.utils

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class SharedViewModel : ViewModel() {

    private val onBottomNavigationViewVisible = MutableLiveData<Boolean>()
    val onBottomNavigationViewVisibilitySelected: LiveData<Boolean>
        get() = onBottomNavigationViewVisible

    //To control the toolbar visibility when showing the GetStartedFragment
    private val onIntroLayoutStarted = MutableLiveData<Boolean>()
    val onStartIntroLayout: LiveData<Boolean>
        get() = onIntroLayoutStarted

    fun setBottomNavigationViewVisibility(isVisible: Boolean) {
        onBottomNavigationViewVisible.value = isVisible
    }

    fun setActiveIntroStarted(isVisible: Boolean) {
        onIntroLayoutStarted.value = isVisible
    }
}